src/main/java/net/minecraftforge/client/model/TRSRTransformation.java

```diff
@@ -68,6 +68,27 @@ public TRSRTransformation(ModelRotation rotation)
     {
         this(rotation.getMatrix());
     }
+    
+    public TRSRTransformation(EnumFacing facing)
+    {
+        this(getMatrix(facing));
+    }
+    
+    public static Matrix4f getMatrix(EnumFacing facing)
+    {
+        Matrix4f ret = new Matrix4f();
+        switch(facing)
+        {
+        case DOWN: ret = ModelRotation.X90_Y0.getMatrix(); break;
+        case UP: ret = ModelRotation.X270_Y0.getMatrix(); break;
+        case NORTH: ret = ModelRotation.X0_Y0.getMatrix(); break;
```

Comment:
TRSRTransformation.identity would probably be better here. And new matrix only needs to be created in the "default" case.